src/Router/Zf2.php

```diff
@@ -54,21 +57,69 @@ public function __construct(TreeRouteStack $router = null)
     public function addRoute(Route $route)
     {
         $path    = $route->getPath();
-        $options = $route->getOptions() ?: [];
+        $options = $route->getOptions();
         $options = array_replace_recursive($options, [
-            'route'   => $route->getPath(),
-            'defaults' => [
-                'middleware' => $route->getMiddleware(),
-            ],
+            'route' => $path
         ]);
-
+        $childRouteName = implode('-', $route->getAllowedMethods());
```

Comment:
If we have a route with `HTTP_METHOD_ANY` I just add to ZF2 route without child. I'm adding a unit test for that case.